File: potd-gfg/Reverse First K elements of Queue (13th Jan 2022)/ReverseFirstKEleOfQueue.java
class GfG {
    // Function to reverse first k elements of a queue.
    public Queue<Integer> modifyQueue(Queue<Integer> q, int k) {
        int n=q.size();
        revK(q,k,1);
        
        for(int i=0;i<n-k;i++){
            q.add(q.remove());
        }
        return q;
    }
    
    public void revK(Queue<Integer> q, int k,int pointer){
        if(pointer>k)
            return;
        int ele=q.remove();
        revK(q,k,pointer+=1);
        q.add(ele);
    }
}

File: potd-gfg/Reverse First K elements of Queue (13th Jan 2022)/ReverseFirstKEleOfQueue.py

'''
Function Arguments :
		@param  : q ( given queue to be used), k(Integer )
		@return : list, just reverse the first k elements of queue and return new queue
'''

#Function to reverse first k elements of a queue.
def modifyQueue(q,k):
    # code here
    temp = list()
    n = len(q)
    
    for _ in range(k):
        temp.append(q.pop(0))
        
    for _ in range(k):
        q.append(temp.pop())
        
    for _ in range(n-k):
        q.append(q.pop(0))
    
    return q
    # return q[:k][::-1] + q[k:]
    # Alternative way to solve this problem in python